internal/kuberesource/mutators.go

```diff
@@ -95,17 +85,84 @@ const (
 	ServiceMeshDisabled serviceMeshMode = "service-mesh-disabled"
 )
 
-// AddServiceMesh adds a service mesh proxy to the resource.
+// AddServiceMesh adds a service mesh proxy to the resource with the proxy
+// configuration given in the object annotations.
 //
 // If the resource does not contain a PodSpec, this function does nothing.
-// This function is not idempotent.
+// This function is idempotent.
 func AddServiceMesh(
 	resource any,
 	serviceMeshProxy *applycorev1.ContainerApplyConfiguration,
-) any {
-	return MapPodSpec(resource, func(spec *applycorev1.PodSpecApplyConfiguration) *applycorev1.PodSpecApplyConfiguration {
+) (res any, retErr error) {
+	res = MapPodSpecWithMeta(resource, func(meta *applymetav1.ObjectMetaApplyConfiguration, spec *applycorev1.PodSpecApplyConfiguration) *applycorev1.PodSpecApplyConfiguration {
+		if spec.RuntimeClassName == nil || !strings.HasPrefix(*spec.RuntimeClassName, "contrast-cc") {
```

Comment:
Not for you to fix here, but a general observation: we should make `contrast-cc` a constant and replace all the string literals.